src/main/java/ay2122s1_cs2103t_w16_2/btbb/storage/JsonAdaptedOrder.java

```diff
@@ -72,6 +96,56 @@ public Order toModelType() throws IllegalValueException {
         }
         final Address modelClientAddress = new Address(clientAddress);
 
-        return new Order(modelClientName, modelClientPhone, modelClientAddress);
+        if (recipeName == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, GenericString.getMessageConstraints("Recipe Name")
+            ));
+        }
+        if (!GenericString.isValidGenericString(recipeName)) {
+            throw new IllegalValueException(GenericString.getMessageConstraints("Recipe Name"));
+        }
+        final GenericString modelRecipeName = new GenericString(recipeName);
+
+        if (recipeIngredients == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, RecipeIngredientList.class.getSimpleName()
+            ));
+        }
+        if (!recipeIngredients.equals("") && !RecipeIngredientList.isValidRecipeIngredientList(recipeIngredients)) {
+            throw new IllegalValueException(RecipeIngredientList.MESSAGE_CONSTRAINTS);
+        }
+        final RecipeIngredientList modelRecipeIngredients = recipeIngredients.equals("")
+                ? new RecipeIngredientList() : new RecipeIngredientList(recipeIngredients);
```

Comment:
If we stored it as a json list, we wouldn't need so many if tags because we can rely on Ingredient.toModelType to ensure that every ingredient field is valid.